mirrord/agent/src/dns.rs

```diff
@@ -110,12 +111,16 @@ impl DnsWorker {
     }
 
     /// Handles the given [`DnsCommand`] in a separate [`tokio::task`].
+    #[tracing::instrument(level = Level::TRACE, skip(self))]
     fn handle_message(&self, message: DnsCommand) {
         let etc_path = self.etc_path.clone();
         let timeout = self.timeout;
         let attempts = self.attempts;
         let lookup_future = async move {
-            let result = Self::do_lookup(etc_path, message.request.node, attempts, timeout).await;
+            let result = Self::do_lookup(etc_path, message.request.node, attempts, timeout)
+                .await
+                .inspect_err(|fail| tracing::error!(?fail, "DNS lookup failed!"));
```

Comment:
```suggestion
                .inspect_err(|fail| tracing::trace!(?fail, "DNS lookup failed!"));
```
I wouldn't log on error level here, resolver build failure should already be logged internally in `do_lookup`. Lookup errors themselves could be spammy